frontend/src/stores/userStore.ts

```diff
@@ -0,0 +1,94 @@
+import { ValidRelationships, type Relationship } from '@/types/Relationship';
+import { getRequest, patchRequest, postRequest } from '@/utils/apiRequests';
+import { defineStore } from 'pinia';
+import type { User } from "../types/User";
+
+export const useUserStore = defineStore("users", {
+
+  //  actions == data definitions
+  state: () => ({
+      allUsers: <User[]>[],
+      me: <User>{},
+  }),
```

Comment:
Wowww I was confused for a bit by this syntax.. But i'm assuming it's this:
`allUsers` of type `array` with elements of type `User`, and you set the default value to an empty array.
`me` is of type `User` and you set the default value to an object.

If yes, you can just resolve this, if no, please explain xD

Reply:
Refactored to a moer readable syntax:
``` typescript
 allUsers: [] as User[],
      me: {} as User,
```